lib/active_admin/router.rb

```diff
@@ -56,6 +56,10 @@ def define_resource_routes(router)
           # Add on the namespace if required
           unless config.namespace.root?
             nested = routes
+
+            # Do not define unnecessary path
+            return if config.resource_name.route_key == 'comments' && ActiveAdmin.application.comments == false
```

Comment:
Line is too long. [111/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.